Server/controllers/authController.js

```diff
@@ -261,6 +262,88 @@ const resetPasswordController = async (req, res, next) => {
     next(error);
   }
 };
+
+/**
+ * Deletes a user and all associated monitors, checks, and alerts.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Object} The response object with success status and message.
+ * @throws {Error} If user validation fails or user is not found in the database.
+ */
+const deleteUserController = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(' ')[1];
+    const decodedToken = jwt.decode(token)
+    const { _id, email } = decodedToken;
+
+    console.log("USERID: ", _id, " Email: ", email)
+
+    const decodedTokenCastedAsRequest = {
+      params: {
+        userId: _id
+      },
+      body: {
+        email
+      }
+    }
```

Comment:
I don't think there's any reason to make a fake request here?  We should just refactor functions that originally wanted a request to now accept an `id` and `email`.

Otherwise this is potentially confusing for new devs.

Let's leave this for now and I'll do a refactor later on.